File: 566/reshape-the-matrix.java
class Solution {
    public int[][] matrixReshape(int[][] mat, int r, int c) {
        if(r==mat.length && c==mat[0].length) return mat;
        if(mat.length*mat[0].length != r*c) return mat;

        int [][] ans = new int[r][c];
        int id=0;
        for(int i=0; i<mat.length; i++){
            for(int j=0; j<mat[0].length; j++){
                int ii = id/c, jj = id%c;
                ans[ii][jj] = mat[i][j];
                id++;
            }
        }
        return ans;
    }
}

File: 566/reshape-the-matrix.py
# LeetCode 566. Reshape the Matrix
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        M, N = len(mat), len(mat[0])
        if M*N != r*c: return mat  # 面積不同、無法正確對應，回傳舊的矩陣

        ans = [[0]*c for _ in range(r)] # 準備好
        for i in range(r):
            ans.append([])  # 
            for j in range(c):
                ii, jj = (i*c+j)//N, (i*c+j)%N
                ans[i][j] = mat[ii][jj]
        return ans
